### PR TITLE
Fix code smells

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Fix `Validate topic subscription with missing topic id` acceptance test
* Fix remaining code smells
* Fix verbose logs in downloader test caused by S3 proxy's logback dependency
* Reduce `Downloader.verifySigsAndDownloadDataFiles()` complexity

**Related issue(s)**:

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — DISMISSED** [review]
Looks good to me.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java [hunk 1]
```diff
@@ -349,72 +346,70 @@ private void verifySigsAndDownloadDataFiles(Multimap<StreamFilename, StreamFileS
...
-                }
+            if (!verifySignatures(signatures)) {
+                log.error("None of the data files could be verified, signatures: {}", signatures);
+            }
 
-                // Ignore signatures that didn't validate or weren't in the majority
-                if (signature.getStatus() != StreamFileSignature.SignatureStatus.CONSENSUS_REACHED) {
-                    continue;
-                }
+            streamVerificationMetric.tag("success", String.valueOf(valid))
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java:353]: now `valid` is always `false`, it should be the return value of `verifySignatures`

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java [hunk 2]
```diff
@@ -197,7 +198,8 @@ private static byte[] recoverAddressFromPubKey(byte[] pubKeyBytes) {
     @SuppressWarnings("java:S1191")
     private static byte[] recoverAddressFromPubKey(LibSecp256k1.secp256k1_pubkey pubKey) {
         final ByteBuffer recoveredFullKey = ByteBuffer.allocate(65);
-        final com.sun.jna.ptr.LongByReference fullKeySize = new com.sun.jna.ptr.LongByReference(recoveredFullKey.limit());
+        final com.sun.jna.ptr.LongByReference fullKeySize =
+                new com.sun.jna.ptr.LongByReference(recoveredFullKey.limit());
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java:202]: nit
```suggestion
        final var fullKeySize = new com.sun.jna.ptr.LongByReference(recoveredFullKey.limit());
```

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/util/Utility.java]: Auto format, but tweaked it to split into another variable.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java:353]: Good catch. Fixed.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**